### PR TITLE
Shipping preview tests

### PR DESCRIPTION
**Pull request by jeffersontuc**
#### What is the purpose of this pull request?

This PR adds tests to shipping preview scenarios.

It depends on [shipping-preview](https://github.com/vtex/shipping-preview/pull/200)

#### What problem is this solving?

With that, besides making `shipping-preview` component proper tested, we ensure that the creation of shipping manager doesn't results on any regression. 

#### How should this be manually tested?

`yarn cypress`

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by marcoskwkm** [comment]
We need to iterate over this, so this should not be blocked by the release of the refactored components. Please create a `refactor/shipping-manager` branch from `master` and set the base branch of this PR to that one. We will use that branch to validate our refactor, just like SP and OS branches with the same name.

File: tests/shipping-preview/models/Delivery - Boleto.model.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+} from '../../../utils/profile-actions'
+import {
+  goToPayment,
+  fillShippingInformation,
+} from '../../../utils/shipping-actions'
+import { payWithBoleto, completePurchase } from '../../../utils/payment-actions'
+import { ACCOUNT_NAMES, SKUS } from '../../../utils/constants'
+
+export default function test(account) {
+  xdescribe(`Delivery - Boleto - ${account}`, () => {
```
Comment (darlenedms) [inline tests/shipping-preview/models/Delivery - Boleto.model.js:15]: You can "unskip" these tests.

File: tests/shipping-preview/models/Delivery - Boleto.model.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        cy.get('#shipping-data')
+          .contains('Mais rápida')
+          .should('be.visible')
+        cy.get('#shipping-data')
+          .contains('Mais econômica')
+          .should('be.visible')
+      }
+      goToPayment()
+      payWithBoleto()
+      completePurchase()
```
Comment (darlenedms) [inline tests/shipping-preview/models/Delivery - Boleto.model.js:50]: I'm not sure if we should proceed to Checkout while testing Shipping Preview (unless we need to make sure its info will be available on Omnishipping), because we may end up with Shipping Preview tests failing because of "finishing purchase" errors.
What do you think? @jeffersontuc @marcoskwkm

Comment (jeffersontuc) [reply tests/shipping-preview/models/Delivery - Boleto.model.js:50]: It makes sense. But in the other way, the intention was not only testing the `shipping-preview`, but also to reuse on `omnishipping` the information (postal code, invoice address) provided to it. And especially now that they are sharing the same `redux` and don't work as fully isolated components anymore. But I agree that a more objective e2e tests provide us more stability. I'm not yet convinced of what path we should take. What do you guys think, @vtex/checkout-ui ?

Comment (marcoskwkm) [reply tests/shipping-preview/models/Delivery - Boleto.model.js:50]: Limiting the scope of Shipping Preview tests just to Shipping Preview itself will greatly improve the reliability of these tests, which I think is very important at the stage we are right now.

I do think we need some tests that include Shipping Preview in the whole purchase flow, especially to test the consistency between Shipping Preview and Omnishipping, but IMO that takes a lower precedence at the moment.

Comment (jeffersontuc) [reply tests/shipping-preview/models/Delivery - Boleto.model.js:50]: It won't take much time to remodel it thought.

File: cypress.json [hunk 3]
```diff
@@ -7,6 +7,6 @@
   "video": true,
   "projectId": "kobqo4",
   "key": "6ecaeca9-8292-49d5-b954-f5d330526703",
-  "pageLoadTimeout": 180000,
-  "defaultCommandTimeout": 15000
+  "pageLoadTimeout": 200000,
+  "defaultCommandTimeout": 20000
 }
```
Comment (marcoskwkm) [inline cypress.json:12]: Why did you need to change these? They are already bigger than they should be...

File: tests/shipping-preview/Delivery Only_Delivery Pickup - Credit card - vtexgame1nolean.test.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import test from './models/Delivery Only_Delivery Pickup - Credit card.model.js'
+
+test('vtexgame1nolean')
```
Comment (marcoskwkm) [inline tests/shipping-preview/Delivery Only_Delivery Pickup - Credit card - vtexgame1nolean.test.js:3]: Why is this the only file that doesn't end with a newline? 🤔

File: tests/shipping-preview/models/Delivery_Scheduled Pickup - Credit card.model.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          .contains('PAC')
+          .should('be.visible')
+      } else {
+        cy.waitAndGet('.srp-content', 3000)
+          .contains('Mais econômica')
+          .should('be.visible')
+      }
+      cy.contains('Retirar').should('be.visible')
+      cy.contains('Loja em Copacabana no Rio de Janeiro')
+        .should('be.visible')
```
Comment (marcoskwkm) [inline tests/shipping-preview/models/Delivery_Scheduled Pickup - Credit card.model.js:34]: Why are you not using the `checkShippingPreviewPickupResult()` function here?

File: utils/payment-actions.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { waitLoad } from '.'
+import { waitLoad } from './'
```
Comment (marcoskwkm) [inline utils/payment-actions.js:1]: Why did you change this file if you're not testing payment?

File: utils/profile-actions.js [hunk 7]
```diff
@@ -4,6 +4,7 @@ import getDocument from './document-generator'
 export function fillEmail(email) {
   cy.get('#cart-to-orderform').click()
   cy.get('#client-pre-email').type(email)
+  cy.wait(3000)
```
Comment (marcoskwkm) [inline utils/profile-actions.js:7]: Why did you change this file if you're not testing profile?

File: utils/shipping-actions.js [hunk 8]
```diff
@@ -19,7 +19,9 @@ function fillGeolocationOmnishipping() {
 }
 
 function fillAddressInformation() {
-  cy.waitAndGet('#ship-number', 3000).type('12')
+  cy.waitAndGet('#ship-postalCode', 1000)
+  cy.waitAndGet('#ship-number', 3000).type('12', { force: true })
+  cy.wait(3000)
```
Comment (marcoskwkm) [inline utils/shipping-actions.js:24]: Why do you need this wait?

File: tests/shipping-preview/models/Delivery Only_Delivery Pickup - Credit card.model.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        cy.get('.srp-content')
+          .contains('PAC')
+          .should('be.visible')
+      } else {
+        cy.waitAndGet('.srp-content', 3000)
+          .contains('Mais econômica')
+          .should('be.visible')
+      }
+
+      choosePickupShippingPreview()
```
Comment (marcoskwkm) [inline tests/shipping-preview/models/Delivery Only_Delivery Pickup - Credit card.model.js:36]: You're missing the part that tests Pickup here.

File: tests/shipping-preview/Delivery Only_Delivery Pickup - Credit card - vtexgame1nolean.test.js (hunk 4 above)
Comment (jeffersontuc) [reply tests/shipping-preview/Delivery Only_Delivery Pickup - Credit card - vtexgame1nolean.test.js]: Oops! Fixed.

File: utils/payment-actions.js (hunk 6 above)
Comment (jeffersontuc) [reply utils/payment-actions.js]: I guess `prettier` wasn't properly setted before.

File: utils/profile-actions.js (hunk 7 above)
Comment (jeffersontuc) [reply utils/profile-actions.js]: I forgot to revert this change when I removed SP tests from the finishing purchase flow.

**Comment by jeffersontuc** [comment]
@marcoskwkm I addressed the changes you requested and the tests are way more reliable now. 

They still can be slightly improved by checking if the postal code or pickup address are presented, for instance. But I think we can go with what we have here.

File: utils/shipping-actions.js [hunk 10]
```diff
@@ -154,3 +160,45 @@ export function selectOtherPickup() {
...
+    if (selector.name) {
+      cy.get('.srp-data')
+        .contains(selector.name)
+        .should('be.visible')
+    }
+    if (selector.text) {
+      cy.get('.srp-data')
+        .contains(selector.text)
+        .should('be.visible')
+    }
```
Comment (marcoskwkm) [inline utils/shipping-actions.js:202]: This is not ideal because you might be matching `name` and `text` in different selectors. But this is fine for now.

File: tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+      if (account === ACCOUNT_NAMES.NO_LEAN) {
+        selectors.push({ name: 'PAC', text: 'Em até 10 dias úteis' })
+      } else {
+        selectors.push({ name: 'Mais econômica', text: 'Em até 10 dias úteis' })
+      }
+
+      checkShippingPreviewResult(selectors)
+      choosePickupShippingPreview()
+
+      selectors.push({ text: 'Pronto em até 11 dias' })
```
[the remark this answers is not in the document]
Comment (jeffersontuc) [reply tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js:38]: Good question. I took it for granted, but you made me think now. Is the `price` a better alternative?

[the remark this answers is not in the document]
Comment (jeffersontuc) [reply tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js:38]: Because the pickup/scheduled delivery selector tend to show only its estimate and price.

[the remark this answers is not in the document]
Comment (marcoskwkm) [reply tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js:38]: Doesn't look like a good idea to me. Tests should not rely on a property of a specific product, but rather only on what SLAs it has available. Imagine the pain it would be if we used prices and one day we had to change the items of the test. We would have to update the values of every test! And that's not even what we want to test in the first place...

File: utils/shipping-actions.js [hunk 12]
```diff
@@ -154,3 +160,43 @@ export function selectOtherPickup() {
...
+      .first()
+      .trigger('mouseover', { force: true })
+
+    cy.get('.pac-item')
+      .first()
+      .click({ force: true })
+  }
+  cy.get('.pkpmodal-points-list .pkpmodal-pickup-point-main').click()
+
+  cy.get('.pkpmodal-details-confirm-btn').click()
```
Comment (marcoskwkm) [inline utils/shipping-actions.js:188]: Do not copy-paste code. Remember why we're doing this refactor in the first place.
Extract this to a function and use it in both here and in `fillPickupAddress`.

Comment (jeffersontuc) [reply utils/shipping-actions.js:188]: You're right! I'll do that.

File: tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+        skus: [
+          SKUS.PICKUP_1_SLA_AND_DELIVERY_MULTIPLE_SLA,
+          SKUS.DELIVERY_MULTIPLE_SLA_AND_PICKUP_AT_PORTO_ALEGRE,
+        ],
+        account,
+      })
+
+      const selectors = []
+
+      cy.contains('Calcular').should('be.visible')
```
Comment (darlenedms) [inline tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js:26]: Why did you choose to check by `Calcular` text instead of `shipping-calculate-link` id?

Comment (jeffersontuc) [reply tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js:26]: This check wasn't meant to this. Actually, it is unnecessary since the subsequent actions are gonna look for the `shipping-calculate-link` id. So, I think I should remove it from the models.

Comment (jeffersontuc) [reply tests/shipping-preview/models/Delivery Only_Delivery Pickup.model.js]: I just removed this check from the models. So I'm asking your review again.

**Comment by jeffersontuc** [comment]
Changed the base branch, so I'm asking reviews again.
